app/middleware/resolve_service.js

```diff
@@ -19,16 +28,22 @@ module.exports = function (req, res, next) {
   }
 
   if (!req.service) {
-    return renderErrorView(req, res, 'You do not have the rights to access this service.')
+    return notAuthorised(req, res)
   }
 
+  delete req.params.externalServiceId
```

Comment:
I think this is `req.service.externalId`

Reply:
no, this is correct - I am deleting the incoming parameter on the request, as we should have now set `req.service` (or thrown an error). Everything after this should now use `req.service.externalId`